lib/blocks.php

```diff
@@ -412,3 +412,36 @@ function gutenberg_block_has_support( $block_type, $feature, $default = false )
 
 	return true === $block_support || is_array( $block_support );
 }
+
+/**
+ * Updates the shape of supports for declaring fontSize and lineHeight.
+ *
+ * @param array $metadata Metadata for registering a block type.
+ * @return array          Metadata for registering a block type with the supports shape updated.
+ */
+function gutenberg_migrate_old_typography_shape( $metadata ) {
+	if ( isset( $metadata['supports'] ) ) {
+		$font_size_support   = _wp_array_get( $metadata['supports'], array( 'fontSize' ), null );
```

Comment:
Would it make sense to migrate the experimental supports as well? I'd think it's not a lot of work and it'd be nice for developers to know what's happening and to maintain back-compatibility for a few versions.